app/controllers/oaisys/pmh_controller.rb

```diff
@@ -17,59 +17,70 @@ class Oaisys::PMHController < Oaisys::ApplicationController
   def bad_verb; end
 
   def identify
-    expect_no_args for_verb: :Identify
-
     respond_to do |format|
       format.xml { render :identify }
     end
   end
 
   def list_sets
-    expect_args for_verb: :ListSets, exclusive: [:resumptionToken]
+    expect_args exclusive: [:resumptionToken]
   end
 
   # TODO: Handle the identifier argument.
   def list_metadata_formats
-    expect_args for_verb: :ListMetadataFormats, optional: [:identifier]
+    expect_args optional: [:identifier]
 
     respond_to do |format|
       format.xml do
-        render :list_metadata_formats, locals: { supported_formats: SUPPORTED_FORMATS }
+        render :list_metadata_formats, locals: { supported_formats: SUPPORTED_FORMATS, parameters: @parameters }
       end
     end
   end
 
   def list_records
-    expect_args for_verb: :ListRecords, required: [:metadataPrefix], optional: [:from, :until, :set],
-                exclusive: [:resumptionToken]
+    expect_args required: [:metadataPrefix], optional: [:from, :until, :set], exclusive: [:resumptionToken]
   end
 
   # get_record is referring to the verb, not a getter.
   # rubocop:disable Naming/AccessorMethodName
   def get_record
-    expect_args for_verb: :GetRecord, required: [:identifier, :metadataPrefix]
+    expect_args required: [:identifier, :metadataPrefix]
   end
   # rubocop:enable Naming/AccessorMethodName
 
+  # TODO: Handle from, until, and resumptionToken arguments.
   def list_identifiers
-    expect_args for_verb: :ListIdentifiers, required: [:metadataPrefix], optional: [:from, :until, :set],
-                exclusive: [:resumptionToken]
+    expect_args required: [:metadataPrefix], optional: [:from, :until, :set], exclusive: [:resumptionToken]
+
+    results = results_from_metadata_prefix
+    results = results.belongs_to_path(params[:set].tr(':', '/')) if params[:set].present?
+    raise Oaisys::NoRecordsMatchError.new(parameters: @parameters.slice(:verb, :metadataPrefix)) if results.empty?
+
+    respond_to do |format|
+      format.xml do
+        render :list_identifiers, locals: { results: results, parameters: @parameters }
+      end
+    end
   end
 
   private
 
-  def expect_args(for_verb:, required: [], optional: [], exclusive: [])
-    arguments = params.except('verb', 'controller', 'action').keys.map(&:to_sym)
-    expected_verb_arguments = required + optional + exclusive
-    unexpected_arguments = (arguments - expected_verb_arguments).present?
-    missing_required_arguments = (required - arguments).present?
-
-    raise Oaisys::BadArgumentError.new(for_verb: for_verb) if unexpected_arguments || missing_required_arguments
+  def expect_args(required: [], optional: [], exclusive: [])
+    ActionController::Parameters.action_on_unpermitted_parameters = :raise
```

Comment:
I've found two places where this could cause issues. There are two calls to params.permit() inside Jupiter, one on line 64 of search_controller.rb and one on line 3 of search_helper.rb.  It looks to me that the only way to have params contain unpermitted parameters in those two cases is if they're manually entered in the search bar (which then an exception would be thrown because of this added line.) I've used logger to search through 2 months of production logs and I couldn't find a single instance where an entry of "Unpermitted parameter: xxx" was made so I _think_ it's safe to make this change.

Reply:
Actually I'm just reading now that "The default value is :log in test and development environments, false otherwise" which if it is false it is not logged so checking the logs didn't help out at all

Reply:
Found some exceptions thrown regarding unpermitted parameters when running tests.

When hitting /search with the header (Content-Type: application/json) or when running `get search_url, as: :json` (which the test does.) the exception is thrown ```
(ActionController::UnpermittedParameters: found unpermitted parameter: :search
)
```. The search param is present but it is an empty hash which is the reason the exception is thrown ("search"=>{}). One way to fix is is just supplying empty an empty string as a search parameter `get search_url, as: :json, params: { search: '' }` I am unsure as to why this happens when requesting a json response but I don't think that is done other than in tests? 

I also found tests failing regarding draft theses... `found unpermitted parameters: :date_created(3i), :date_created(2i)` happens when trying to move to step two of the wizard for a thesis. Those parameters are created on line 56 and 58 of draft_actions.rb and are not in the permitted_attributes array in draft_thesis_policy.rb. I've added a check for `@draft.is_a?(DraftItem)` before setting the month and day automatically because date_created doesn't pertain to theses.

I think these are appropriate fixes, we could still revert to checking the parameters manually if you think there could be too many complications. I've included my changes in this PR: https://github.com/ualbertalib/jupiter/pull/1361